sdk/digitaltwins/digital-twins-parser/src/parser/helpers.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/* eslint-disable valid-jsdoc */
+/* eslint-disable guard-for-in */
+/* eslint-disable no-empty */
+/* eslint-disable no-unused-vars */
+/* eslint-disable sort-imports */
+
+import { EntityInfo } from "./internal";
+import { LanguageStringType } from "./internal";
+/**
+ * A static class that holds various helper functions.
+ **/
+export class Helpers {
+  static areListsIdEqual(list1: EntityInfo[], list2: EntityInfo[]): boolean {
+    if (list1.length !== list2.length) {
+      return false;
+    }
+
+    for (let i = 0; i < list1.length; i++) {
+      if (list1[i].id !== list2[i].id) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  static areListsIdOrLiteralEqual(list1: EntityInfo[], list2: EntityInfo[]): boolean {
+    if (list1.length !== list2.length) {
+      return false;
+    }
+
+    for (let i = 0; i < list1.length; i++) {
+      if (list1[i].id !== undefined && list2[i].id !== undefined) {
+        if (list1[i].id !== list2[i].id) {
+          return false;
+        }
+      } else {
+        const string1 = JSON.stringify(list1[i]);
```

Comment:
seems like a pretty heavy way to determine deep equality

Reply:
is there a lighter way for deep equality? `v8.deserialize(v8.serialize(list1[i]))?`

Reply:
So it's not a trivial problem, but if you're willing to take an external dep, this implementation appears to be more optimized: https://www.npmjs.com/package/deep-eql

Reply:
for now we will hold off on changing this. I think the v8 method seems like the most linear way to approach deep equality without an external dependency.